eclair-core/src/main/scala/fr/acinq/eclair/message/Postman.scala

```diff
@@ -146,40 +148,56 @@ private class SendingMessage(nodeParams: NodeParams,
   def start(): Behavior[Command] = {
     Behaviors.receiveMessagePartial {
       case SendMessage =>
-        val targetNodeId = destination match {
-          case OnionMessages.BlindedPath(route) => route.introductionNodeId
-          case OnionMessages.Recipient(nodeId, _, _, _) => nodeId
-        }
-        routingStrategy match {
-          case RoutingStrategy.UseRoute(intermediateNodes) => sendToRoute(intermediateNodes, targetNodeId)
-          case RoutingStrategy.FindRoute if targetNodeId == nodeParams.nodeId =>
-            context.self ! WrappedMessageRouteResponse(MessageRoute(Nil, targetNodeId))
-            waitForRouteFromRouter()
-          case RoutingStrategy.FindRoute =>
-            router ! Router.MessageRouteRequest(context.messageAdapter(WrappedMessageRouteResponse), nodeParams.nodeId, targetNodeId, Set.empty)
-            waitForRouteFromRouter()
+        contactInfo match {
+          case compact: OfferTypes.CompactBlindedPath =>
+            router ! Router.GetNodeId(context.messageAdapter(WrappedNodeIdResponse), compact.introductionNode.scid, compact.introductionNode.isNode1)
+            waitForNodeId(compact)
+          case OfferTypes.BlindedPath(route) => sendToDestination(OnionMessages.BlindedPath(route))
+          case OfferTypes.RecipientNodeId(nodeId) => sendToDestination(OnionMessages.Recipient(nodeId, None))
         }
     }
   }
 
-  private def waitForRouteFromRouter(): Behavior[Command] = {
+  private def waitForNodeId(compactBlindedPath: CompactBlindedPath): Behavior[Command] = {
+    Behaviors.receiveMessagePartial {
+      case WrappedNodeIdResponse(None) =>
+        replyTo ! Postman.MessageFailed("Unknown target")
```

Comment:
nit: it would be useful to include the scid in the error message:

```suggestion
        replyTo ! Postman.MessageFailed(s"Could not resolve introduction node for compact blinded path (scid=${compactBlindedPath.introductionNode.scid.toCoordinatesString})")
```